### DISCUSSION TITLE
PSR-7 HTTP bridge request body issue

### DISCUSSION THREAD

**Discussion by rguennichi** [symfony/symfony#47017] [Q&A] [answered]
Hi everyone, does anyone experienced some strange behaviour with PSR-7 HTTP bridge provided by Symfony: https://symfony.com/doc/current/components/psr7.html
```php
$psrRequest = $psrHttpFactory->createRequest($symfonyRequest);
```
returns an instance of `Psr\Http\Message\ServerRequestInterface` , the problem here is related to the `$psrRequest->getBody()` which returns only the value of the `$symfonyRequest->getContent(resource: true)` , I don’t know, but I’m thinking this should not be the case, as the `getBody()` of a PSR request should return the entire request body stream, by appending $_POST params, $_FILES and request content resources, can anyone explain why it’s not the case? maybe I’m misunderstanding the concept of Request/ServerRequest body

**Comment by MatTheCat** [comment, marked as answer]
https://www.php.net/manual/en/wrappers.php.php#wrappers.php.input

> `php://input` is not available with `enctype="multipart/form-data"`.

So you wouldn’t access uploaded files anyway.

See also https://www.php-fig.org/psr/psr-7/meta/#rationale-for-serverrequestinterface

**Reply by rguennichi** [reply]
Thanks for the answer, I already created a library that helps creating PSR request from PSR server request object https://github.com/rguennichi/psr7-request-creator

**Reply by MatTheCat** [reply]
Great! Can I ask what your use case was? `ServerRequestInterface` seems nicer to interact with than `RequestInterface`.

**Reply by rguennichi** [reply]
We used that library in one of our Symfony's project where the app itself is considered as a **router**/**proxy**/**bridge** to communicate with our internal apps via HTTP, so basically it transform the current server request object to a simple request object that we can use to call clients with different endpoints via Httplug.
